x/pki/keeper/msg_server_remove_noc_x_509_root_cert.go

```diff
@@ -0,0 +1,114 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	pkitypes "github.com/zigbee-alliance/distributed-compliance-ledger/types/pki"
+	dclauthtypes "github.com/zigbee-alliance/distributed-compliance-ledger/x/dclauth/types"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/pki/types"
+)
+
+func (k msgServer) RemoveNocX509RootCert(goCtx context.Context, msg *types.MsgRemoveNocX509RootCert) (*types.MsgRemoveNocX509RootCertResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, pkitypes.NewErrInvalidAddress(err)
+	}
+
+	// check if signer has vendor role
+	if !k.dclauthKeeper.HasRole(ctx, signerAddr, dclauthtypes.Vendor) {
+		return nil, pkitypes.NewErrUnauthorizedRole("MsgRemoveNocX509RootCert", dclauthtypes.Vendor)
+	}
+
+	signerAccount, _ := k.dclauthKeeper.GetAccountO(ctx, signerAddr)
+	accountVid := signerAccount.VendorID
+
+	nocCerts, foundActive := k.GetNocRootCertificatesByVidAndSkid(ctx, accountVid, msg.SubjectKeyId)
+	revCerts, foundRevoked := k.GetRevokedNocRootCertificates(ctx, msg.Subject, msg.SubjectKeyId)
+	certificates := nocCerts.Certs
+	certificates = append(certificates, revCerts.Certs...)
+	if len(certificates) == 0 {
+		return nil, pkitypes.NewErrCertificateDoesNotExist(msg.Subject, msg.SubjectKeyId)
+	}
+
+	certID := types.CertificateIdentifier{
+		Subject:      msg.Subject,
+		SubjectKeyId: msg.SubjectKeyId,
+	}
+
+	if msg.SerialNumber != "" {
+		certBySerialNumber, found := findCertificate(msg.SerialNumber, &certificates)
+		if !found {
+			return nil, pkitypes.NewErrCertificateBySerialNumberDoesNotExist(msg.Subject, msg.SubjectKeyId, msg.SerialNumber)
+		}
+
+		// remove from subject with serialNumber map
+		k.RemoveUniqueCertificate(ctx, certBySerialNumber.Subject, certBySerialNumber.SerialNumber)
+
+		if foundActive {
+			// Remove from Approved lists
+			aprCerts, _ := k.GetApprovedCertificates(ctx, msg.Subject, msg.SubjectKeyId)
+			removeCertFromList(certBySerialNumber.Issuer, certBySerialNumber.SerialNumber, &aprCerts.Certs)
+			k.removeApprovedX509Cert(ctx, certID, &aprCerts, msg.SerialNumber)
+
+			// Remove from NOC lists
+			removeCertFromList(certBySerialNumber.Issuer, certBySerialNumber.SerialNumber, &nocCerts.Certs)
+			k._removeNocX509RootCert(ctx, certID, &nocCerts, msg.SerialNumber)
```

Comment:
We can just call `k.RemoveNocRootCertificateByVidSubjectSkidAndSerialNumber` here